src/content/docs/docs/guides/react.mdx

```diff
@@ -0,0 +1,774 @@
+  todosCollection.insertOne(unpersistedTodo);
+
+  setTodoText("");
+}
+```
+
+The main line removed is `setTodos([...todos, { id, ...unpersistedTodo }]);`. The function no longer needs to be async as you don't need to wait for the ID from the database and is updated accordingly.
+
+To test the realtime updates, open your app in a private window while keeping the original window open. When you make changes in one window, they will appear in the other in realtime.
+
+Realtime syncing is enabled, but it only handles add todo events. In the next step, you will implement the functionality to toggle the completion status of todos and update your subscribe listener with an `onChange` handler.
+
+## Step 10 — Toggling Todo Completion
+
+In this step, you'll implement functionality to toggle the completion status of todos.
+
+First, add an `onChange` handler to your subscribe listener:
+
+```tsx title="App.tsx" ins={5-9}
+todosCollection.subscribeAll(
+  {},
+  {
+    onAdd: (doc) => {...},
+    onChange: (_, newDoc) => {
+      setTodos((todos) =>
+        todos.map((todo) => (todo.id === newDoc.id ? newDoc : todo))
+      );
+    },
+  },
+);
+```
+
+The `onChange` handler updates the corresponding todo in your local state. Note that the first parameter in the `onChange` handler is `newDoc`, which you don't need. An underscore is used as the parameter name to indicate the variable is unused and to avoid a React build error.
+
+Next, update your template:
+
+```tsx title="App.tsx" ins={3-6,8-10}
+<div className="todos">
+  {reversedTodos.map((todo) => (
+    <div
+      className={`card ${todo.completed && "strikethrough"}`}
+      key={todo.id}
+    >
+      <div className="todo-text">{todo.text}</div>
+      <button onClick={() => toggleTodoCompletion(todo)}>
+        Mark as {todo.completed ? "Incomplete" : "Completed"}
+      </button>
+    </div>
+  ))}
+</div>
+```
+
+In these changes, you add a `strikethrough` class to a todo when it's complete. The `strikethrough` style is already in `index.css`.
+
+And add the `toggleTodoCompletion` click event handler below `onSubmit`:
+
+```tsx title="App.tsx" ins={3-5}
+function onSubmit(event: FormEvent) {...}
+
+function toggleTodoCompletion(todo: Todo) {
+  todosCollection.updateOne(todo.id, { completed: !todo.completed });
+}
+```
+
+That function updates the `completed` value for a todo, setting it to the opposite of its current state.
+
+You can now mark todos as completed or incomplete. In the next step, you will add the functionality to delete todos.
+
+## Step 11 — Deleting a Todo
+
+In this step, you will implement the functionality to delete todos.
+
+First, add an `onDelete` handler to your subscribe listener:
+
+```tsx title="App.tsx" ins={6-8}
+todosCollection.subscribeAll(
+  {},
+  {
+    onAdd: (doc) => {...},
+    onChange: (_, newDoc) => {...},
+    onDelete: (doc) => {
+      setTodos((todos) => todos.filter((todo) => todo.id !== doc.id));
```

Comment:
`onChange` and `onDelete` use a function in `setTodos`. I guess because you could quickly change multiple todos before it re-renders? In that case, I would argue you should also use a function in `onAdd`.

Reply:
I do use a function for `onAdd`, right?

https://github.com/bzr-sys/bazaar-website/blob/d7cb726f0faa1ff580ef8a218c2729c931f37c98/src/content/docs/docs/guides/react.mdx?plain=1#L458-L463

My thinking was actually the same, but the other way around. I need a function for `onAdd` to avoid adding duplicates, so `onChange` and `onDelete` should use a function too.

Reply:
My bad, must have been confused.